tensorflow/java/src/gen/cc/java_defs.h

```diff
@@ -0,0 +1,266 @@
+  bool IsCollection() const {
+    return name_ == "List" || name_ == "Iterable";
+  }
+  /// Returns true if this instance is a wildcard (<?>)
+  bool IsWildcard() const {
+    return kind_ == GENERIC && name_.empty();
+  }
+  bool operator==(const Type& type) const {
+      return name_ == type.name_ && package_ == type.package_;
+  }
+  bool operator!=(const Type& type) const { return !(*this == type); }
+
+ protected:
+  Type(Kind kind, const string& name, const string& package)
+    : kind_(kind), name_(name), package_(package) {}
+
+ private:
+  Kind kind_;
+  string name_;
+  string package_;
+  string descr_;
+  std::vector<Type> params_;
+  std::vector<Annotation> annotations_;
+  std::deque<Type> supertypes_;
+};
+
+/// \brief Definition of a Java annotation
+///
+/// This class only defines the usage of an annotation in a specific context,
+/// giving optionally a set of attributes to initialize.
+class Annotation : public Type {
+ public:
+  static Annotation Of(const string& type_name, const string& package = "") {
+    return Annotation(type_name, package);
+  }
+  const string& attrs() const { return attrs_; }
+  Annotation& attrs(const string& attrs) { attrs_ = attrs; return *this; }
+
+ private:
+  string attrs_;
+
+  Annotation(const string& name, const string& package)
+    : Type(Kind::ANNOTATION, name, package) {}
+};
+
+/// \brief A definition of a Java variable
+///
+/// This class defines an instance of a type, which could be documented.
+class Variable {
+ public:
+  static Variable Of(const string& name, const Type& type) {
+    return Variable(name, type, false);
+  }
+  static Variable VarArg(const string& name, const Type& type) {
+    return Variable(name, type, true);
+  }
+  const string& name() const { return name_; }
+  const Type& type() const { return type_; }
+  bool variadic() const { return variadic_; }
+  const string& descr() const { return descr_; }
+  Variable& descr(const string& descr) { descr_ = descr; return *this; }
+
+ private:
+  string name_;
+  Type type_;
+  bool variadic_;
+  string descr_;
+
+  Variable(const string& name, const Type& type, bool variadic)
+    : name_(name), type_(type), variadic_(variadic) {}
+};
+
+/// \brief A definition of a Java class method
+///
+/// This class defines the signature of a method, including its name, return
+/// type and arguments.
+class Method {
+ public:
+  static Method Of(const string& name, const Type& return_type) {
+    return Method(name, return_type, false);
+  }
+  static Method ConstructorFor(const Type& clazz) {
+    return Method(clazz.name(), clazz, true);
+  }
+  bool constructor() const { return constructor_; }
+  const string& name() const { return name_; }
+  const Type& ret_type() const { return ret_type_; }
+  const string& descr() const { return descr_; }
+  Method& descr(const string& descr) { descr_ = descr; return *this; }
+  const string& ret_descr() const { return ret_descr_; }
```

Comment:
Use `description` instead of `descr` and `return_description()` instead of `ret_descr`?
(And `return_type()` instead of `ret_type()` etc.)